mbed_targets/mbed_targets.py

```diff
@@ -126,13 +120,27 @@ class MbedTargets(Set):
 
     @classmethod
     def from_offline_database(cls) -> "MbedTargets":
-        """Initialise with the offline target database."""
-        return cls(MbedTarget(**t) for t in target_database.get_offline_target_data())
+        """Initialise with the offline target database.
+
+        Raises:
+            TargetDataNotFound: Could not retrieve data from the target database.
+        """
+        try:
+            return cls(MbedTarget(**t) for t in target_database.get_offline_target_data())
+        except TargetDatabaseError as tgt_db_err:
```

Comment:
I don't really understand why we need to change the exception,

Reply:
I didn't want to expose internal exceptions to the user. I was discussing this on Slack with @muchzill4 and he asked similar questions, so maybe I should make the `TargetDatabaseError` public? Although all the internal exceptions derive from it so I thought that was weird, but I guess it's the same for `ToolsError`.

Reply:
64441d3